libminifi/include/utils/MinifiConcurrentQueue.h

```diff
@@ -29,11 +29,12 @@ namespace nifi {
 namespace minifi {
 namespace utils {
 
+
+// Provides a queue API and guarantees no race conditions in case of multiple producers and consumers.
```

Comment:
I would add:
- Guarantees that the elements are dequeued in the order of insertion

Reply:
Added